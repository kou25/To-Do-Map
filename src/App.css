
body{
    background-image: linear-gradient(120deg, #fdfbfb 0%, #ebedee 100%);
    
}


.nav-head{
    top: 0;
    left: 0;
    width: 100%;
    background: orange;
    height: 73px;
    -webkit-box-shadow: -1px 4px 12px 1px rgba(0,0,0,0.5);
-moz-box-shadow: -1px 4px 12px 1px rgba(0,0,0,0.5);
box-shadow: -1px 4px 12px 1px rgba(0,0,0,0.5);
}

.navigation{
    display: flex;
    height: 100%;
    align-items: center;
    padding: 0 2rem;
}
.nav-logos{
    margin-left: 1rem;
}
.nav-logos a{
    color: white;
    text-decoration: none; 
    font-size: 1.5rem;
    font-family: cursive;   
}

.items a{
    color: white;
    text-decoration:none ;
    font-family: cursive;
}

.spacer{
    flex: 1;
}


.items ul{
    list-style: none;
    margin: 0;
    padding: 0;
    display: flex;
}
.items li{
    padding: 0 0.5rem;
}

.items a:hover, .items a:active{
    color: black
}

.nav-logos a:hover, .nav-logos a:active{
    color: black
}

#divhover h5:hover, #divhover h5:active{
    color: orange;
}

#img{
    background-image: url("./components/Landing/Landing.jpg");
    height: 90vh;
    width: 100%;
    background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
}
tbody:nth-of-type(even) {
    background-color: #e9e9f9 ;
}

.right {
    margin-top: 40px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: 105%;
    background-color: orange;
    width: 100%;
    position: absolute;
    right: -15%;
    border-radius: 25px;
    z-index: -2;
    box-shadow: 0px 0px 12px 6px rgba(15, 15, 15, 0.281);
  }

  .rights {
    margin-top: 30px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: 100%;
    background-color: orange;
    width: 125%;
    position: absolute;
    right: -14%;
    border-radius: 25px;
    z-index: -2;
    box-shadow: 0px 0px 12px 6px rgba(15, 15, 15, 0.281);
  }

  
