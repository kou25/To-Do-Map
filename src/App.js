import React, { Component } from 'react';
import './App.css';
import {Provider} from 'react-redux'
import Navbar from './components/navbar/Navbar';
import Sidebar from './components/Sidebar/Sidebar';
import Backdrop from './components/Sidebar/Backdrop';
import Dashboard from './components/dashboard/Dashboard';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Add from './containers/add/Add';
import View from './containers/view/View';
import store from './store/store'
import Landing from './components/Landing/Landing';
import Edit from './containers/edit/Edit';
import More from './containers/more/More';
import Signin from './components/login/Signin';
import SignUp from './components/login/SignUp';
import Forget from './components/login/Forget';
import Map from './components/map/Map'



class App extends Component {
  state={
    sidebarOpen: false,
  };

  toggleClickHandler=()=>{
    this.setState((prevState)=>{
      return{sidebarOpen: !prevState.sidebarOpen};
    });
  }
  backdropClickHadler=()=>{
    this.setState({sidebarOpen: false})
  }

  

  render(){
    let backdrop;

    if (this.state.sidebarOpen){
      backdrop=<Backdrop click={this.backdropClickHadler}/>
    }
  return (
    <Provider store={store}>
    <Router>
    <div style={{height: '100%'}}>
      <Navbar clickHandler={this.toggleClickHandler}/>
      <Sidebar show={this.state.sidebarOpen}/>
      {backdrop}
      <main style={{marginTop: '2px'}}>
        <Switch>
        <Route exact path="/signin" component={Signin} />
        <Route exact path="/signup" component={SignUp} />
        <Route exact path="/forget" component={Forget} />
        <Route exact path="/" component={Landing} />
        <Route exact path="/dashboard" component={Dashboard}/>
        <Route exact path="/add" component={Add} />
        <Route exact path="/view" component={View} />
        <Route exact path="/map" component={Map} />
        <Route exact path="/todo/edit/:id" component={Edit} />
        <Route exact path="/more/:id" component={More} />
        </Switch>
      </main>
    </div>
    </Router>
    </Provider>
  );
}
}

export default App;
