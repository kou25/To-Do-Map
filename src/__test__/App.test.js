import React from 'react';
import ReactDom from 'react-dom'
import App from '../App';
import renderer from 'react-test-renderer'
import {cleanup} from '@testing-library/react'

afterEach(cleanup)
it("render without crashing",()=>{
  const div= document.createElement("div");
  ReactDom.render(<App/>,div);
  ReactDom.unmountComponentAtNode(div);
})

it('matches snapshot', ()=>{
  const tree=renderer.create(
  <App/>).toJSON()
  expect(tree).toMatchSnapshot()
})
