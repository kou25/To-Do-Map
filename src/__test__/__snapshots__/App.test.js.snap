// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  style={
    Object {
      "height": "100%",
    }
  }
>
  <header
    className="nav-head"
  >
    <nav
      className="navigation"
    >
      <div>
        <button
          className="button"
          onClick={[Function]}
        >
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
        </button>
      </div>
      <div
        className="nav-logos"
      >
        <a
          data-testid="img"
          href="/landing"
          style={
            Object {
              "textShadow": "1px 1px 2px black",
            }
          }
        >
          <img
            alt="logo"
            src="note.png"
            style={
              Object {
                "height": "70px",
              }
            }
          />
          TO DO
        </a>
      </div>
      <div
        className="spacer"
      />
    </nav>
  </header>
  <nav
    className="sidebar"
  >
    <div
      className="pro"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          src="profile.png"
          style={
            Object {
              "height": "70px",
            }
          }
        />
        <span
          style={
            Object {
              "marginLeft": "40px",
            }
          }
        >
          Welcome
        </span>
      </a>
    </div>
    <hr
      style={
        Object {
          "backgroundColor": "orange",
          "height": "1.2px",
          "margin": "3px 5px",
        }
      }
    />
    <ul>
      <li
        data-testid="list"
      >
        <i
          aria-hidden="true"
          className="fa fa-line-chart"
        />
        <a
          href="/dashboard"
          onClick={[Function]}
        >
          Dashboard
        </a>
        <i
          aria-hidden="true"
          className="fa fa-arrow-circle-right"
          style={
            Object {
              "marginLeft": "120px",
            }
          }
        />
      </li>
      <li
        data-testid="list1"
      >
        <i
          aria-hidden="true"
          className="fa fa-plus-circle"
        />
        <a
          href="/add"
          onClick={[Function]}
        >
          Add To-DO
        </a>
        <i
          aria-hidden="true"
          className="fa fa-arrow-circle-right"
          style={
            Object {
              "marginLeft": "110px",
            }
          }
        />
      </li>
      <li
        data-testid="list2"
      >
        <i
          aria-hidden="true"
          className="fa fa-eye"
        />
        <a
          href="/view"
          onClick={[Function]}
        >
          View To-DO
        </a>
        <i
          aria-hidden="true"
          className="fa fa-arrow-circle-right"
          style={
            Object {
              "marginLeft": "103px",
            }
          }
        />
      </li>
      <li
        data-testid="list3"
      >
        <i
          aria-hidden="true"
          className="fa fa-map-marker"
        />
        <a
          href="/map"
          onClick={[Function]}
        >
          View Map
        </a>
        <i
          aria-hidden="true"
          className="fa fa-arrow-circle-right"
          style={
            Object {
              "marginLeft": "132px",
            }
          }
        />
      </li>
    </ul>
  </nav>
  <main
    style={
      Object {
        "marginTop": "2px",
      }
    }
  >
    <div
      className="container"
    >
      <div
        className="d-sm-flex align-items-center justify-content-between mb-4"
        data-testid="heading"
      >
        <h1
          style={
            Object {
              "color": "#8B4513",
              "fontFamily": "cursive",
              "margin": "30px 30%",
            }
          }
        >
          Sign In
          <i
            className="fa fa-sign-in"
            style={
              Object {
                "color": "#8B4513",
                "marginLeft": "20px",
              }
            }
          />
        </h1>
      </div>
      <form
        onSubmit={[Function]}
        style={
          Object {
            "fontFamily": "cursive",
            "marginLeft": "10%",
            "width": "60%",
          }
        }
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="exampleInputEmail1"
          >
            Email address
          </label>
          <input
            className="form-control"
            id="email"
            onChange={[Function]}
            placeholder="Enter email"
            required={true}
            type="email"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="exampleInputPassword1"
          >
            Password
          </label>
          <input
            className="form-control"
            id="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
          />
        </div>
        <button
          className="btn btn-warning mr-2 shadow"
          data-testid="button"
          type="submit"
        >
          Sign In
        </button>
        <a
          href="/signup"
          onClick={[Function]}
        >
          <button
            className="btn btn-primary mr-2 shadow"
            data-testid="button1"
            type="submit"
          >
            Sign Up
          </button>
        </a>
        <a
          href="/forget"
          onClick={[Function]}
        >
          <button
            className="btn btn-danger shadow"
            data-testid="button2"
            type="submit"
          >
            Forget Password
          </button>
        </a>
      </form>
    </div>
  </main>
</div>
`;
